src/screens/TransactionDetail.js

```diff
@@ -37,6 +38,45 @@ function TransactionDetail() {
   const [spentOutputs, setSpentOutputs] = useState(null);
   /* confirmationData {Object} Confirmation data of loaded transaction received from the server */
   const [confirmationData, setConfirmationData] = useState(null);
+  /* alertPageRefreshed {boolean} If should show an info alert to indicate the page was refreshed */
+  const [alertPageRefreshed, setAlertPageRefreshed] = useState(false);
+
+  /**
+   * Called when 'network' ws message arrives
+   * If it's a new block, show an info alert to refresh the page
+   *
+   * @param {Object} wsData Data from websocket
+   */
+  const handleWebsocket = useCallback(async wsData => {
+    // Ignore events not related to this screen
+    if (wsData.type !== 'network:new_tx_accepted') {
+      return;
+    }
+
+    // If the new transaction isn't a block, it cannot have confirmed our tx
+    if (!hathorLib.transactionUtils.isBlock(wsData)) {
+      return;
+    }
+
+    const txData = await txApi.getTransaction(txUid).catch(err => console.log(err));
+    if (!txData?.success) {
+      // The transaction couldn't be fetched, possibly a network error. Just return and wait for the next block.
+      return;
+    }
+
+    if (!txData.meta?.first_block) {
+      // The block did not confirm our transaction yet, just return and wait for the next one
+      return;
+    }
+
+    // Our transaction now has a first block, so we can update its data
+    setAlertPageRefreshed(true);
+
+    // Remove the listener since we only need to detect the first block once
+    WebSocketHandler.removeListener('network', handleWebsocket);
+
+    updateTxInfo(txUid).catch(e => console.error(e));
```

Comment:
If you call this method, you are sending a new request to fetch the data you already have. Maybe you could create a method `fetchAndUpdateTxInfo`

```
const txData = await txApi.getTransaction(id);
updateTxInfo(txData)
```

And the method `updateTxInfo` just receives the data already fetched from the node and update the state.

What do you think?